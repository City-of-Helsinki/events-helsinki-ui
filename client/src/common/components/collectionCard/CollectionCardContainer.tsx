import classNames from "classnames";
import React from "react";

import CollectionCard, {
  CollectionCardSize,
  CollectionCardType
} from "./CollectionCard";
import styles from "./collectionCardContainer.module.scss";

interface CollectionCardContainerProps {
  cards: CollectionCardType[];
  size: CollectionCardSize;
}

const CollectionCardContainer: React.FC<CollectionCardContainerProps> = ({
  cards,
  size
}) => {
  return (
    <div
      className={classNames(
        styles.collectionCardContainer,
        styles[`${size}Size`]
      )}
    >
      {cards.map(card => {
        return (
          <CollectionCard
            count={card.count}
            description={card.description}
            id={card.id}
            size={size}
            title={card.title}
          />
        );
      })}
    </div>
  );
};

export default CollectionCardContainer;
