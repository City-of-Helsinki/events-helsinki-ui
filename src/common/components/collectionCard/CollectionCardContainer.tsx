import classNames from "classnames";
import React from "react";

import { CollectionDetails } from "../../../generated/graphql";
import useLocale from "../../../hooks/useLocale";
import getLocalisedString from "../../../util/getLocalisedString";
import CollectionCard, { CollectionCardSize } from "./CollectionCard";
import styles from "./collectionCardContainer.module.scss";

export type CollectionCardListLayout = "sm" | "md" | "mdAndSm" | "lg";

interface CollectionCardContainerProps {
  collections: CollectionDetails[];
  layout: CollectionCardListLayout;
}

const CollectionCardContainer: React.FC<CollectionCardContainerProps> = ({
  collections,
  layout
}) => {
  const locale = useLocale();
  return (
    <div
      className={classNames(
        styles.collectionCardContainer,
        styles[`${layout}Layout`]
      )}
    >
      {collections.map((collection, index) => {
        const getCardSize = (): CollectionCardSize => {
          switch (layout) {
            case "sm":
              return "sm";
            case "md":
              return "md";
            case "mdAndSm":
              if (index % 5 === 0 || index % 5 === 4) {
                return "md";
              }
              return "sm";
            case "lg":
              return "lg";
          }
        };
        const size = getCardSize();
        return (
          <CollectionCard
            key={index}
            count={collection.curatedEvents.length}
            description={getLocalisedString(collection.description, locale)}
            id={collection.id}
            size={size}
            showDescription={size === "lg"}
            title={getLocalisedString(collection.title, locale)}
          />
        );
      })}
    </div>
  );
};

export default CollectionCardContainer;
