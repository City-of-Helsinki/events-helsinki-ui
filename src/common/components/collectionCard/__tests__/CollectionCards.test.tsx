import { render } from '@testing-library/react';
import * as React from 'react';
import { MemoryRouter } from 'react-router';

import mockCollection from '../../../../domain/collection/__mocks__/collection';
import CollectionCardContainer from '../CollectionCardContainer';

const collections = ['1', '2', '3', '4', '5', '6', '7'].map(id => ({
  ...mockCollection,
  id,
}));

it('matches snapshot', () => {
  const { container, rerender } = render(
    <MemoryRouter>
      <CollectionCardContainer collections={collections} layout="lg" />
    </MemoryRouter>
  );

  expect(container.firstChild).toMatchSnapshot();

  rerender(
    <MemoryRouter>
      <CollectionCardContainer collections={collections} layout="md" />
    </MemoryRouter>
  );

  expect(container.firstChild).toMatchSnapshot();

  rerender(
    <MemoryRouter>
      <CollectionCardContainer collections={collections} layout="mdAndSm" />
    </MemoryRouter>
  );

  expect(container.firstChild).toMatchSnapshot();

  rerender(
    <MemoryRouter>
      <CollectionCardContainer collections={collections} layout="sm" />
    </MemoryRouter>
  );

  expect(container.firstChild).toMatchSnapshot();
});
