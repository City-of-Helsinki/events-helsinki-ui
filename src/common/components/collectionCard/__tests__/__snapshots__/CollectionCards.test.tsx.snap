// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionCardContainer matches snapshot 1`] = `
<div
  className="collectionCardContainer lgSize"
>
  <div
    className="collectionCardWrapper lgSize"
  >
    <a
      aria-hidden={true}
      aria-label="Siirry tapahtumaan: "
      className="imageWrapper"
      href="/fi/collection/1"
      onClick={[Function]}
      tabIndex={-1}
    />
    <div
      className="textWrapper"
    >
      <div
        className="countWrapper"
      >
        <div
          className="count"
        >
          120 kpl
        </div>
      </div>
      <div
        className="titleWrapper"
      >
        <a
          aria-hidden={true}
          className="title"
          href="/fi/collection/1"
          onClick={[Function]}
          tabIndex={-1}
        >
          Title 1
        </a>
        <div
          className="srOnly"
        >
          Title 1
        </div>
        <div
          className="description"
        >
          Lorem ipsum 1
        </div>
      </div>
      <div
        className="linkWrapper"
      >
        <a
          aria-label="Siirry tapahtumaan: "
          className="iconLink default"
          href="/fi/collection/1"
          onClick={[Function]}
        >
          <div
            className="iconWrapper"
          >
            <svg
              className=""
              height="100%"
              viewBox="0 0 24 24"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <polygon
                  points="13.3706831 2 11.5237192 3.84696395 18.0000474 10.3242884 2 10.3242884 2 12.937334 17.9990512 12.937334 11.5237192 19.412666 13.3706831 21.2586338 23 11.6293169"
                />
              </g>
            </svg>
          </div>
        </a>
      </div>
    </div>
  </div>
  <div
    className="collectionCardWrapper lgSize"
  >
    <a
      aria-hidden={true}
      aria-label="Siirry tapahtumaan: "
      className="imageWrapper"
      href="/fi/collection/2"
      onClick={[Function]}
      tabIndex={-1}
    />
    <div
      className="textWrapper"
    >
      <div
        className="countWrapper"
      >
        <div
          className="count"
        >
          120 kpl
        </div>
      </div>
      <div
        className="titleWrapper"
      >
        <a
          aria-hidden={true}
          className="title"
          href="/fi/collection/2"
          onClick={[Function]}
          tabIndex={-1}
        >
          Title 2
        </a>
        <div
          className="srOnly"
        >
          Title 2
        </div>
        <div
          className="description"
        >
          Lorem ipsum 2
        </div>
      </div>
      <div
        className="linkWrapper"
      >
        <a
          aria-label="Siirry tapahtumaan: "
          className="iconLink default"
          href="/fi/collection/2"
          onClick={[Function]}
        >
          <div
            className="iconWrapper"
          >
            <svg
              className=""
              height="100%"
              viewBox="0 0 24 24"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <polygon
                  points="13.3706831 2 11.5237192 3.84696395 18.0000474 10.3242884 2 10.3242884 2 12.937334 17.9990512 12.937334 11.5237192 19.412666 13.3706831 21.2586338 23 11.6293169"
                />
              </g>
            </svg>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
